gdx/src/com/badlogic/gdx/utils/ScreenCapturer.java

```diff
@@ -0,0 +1,185 @@
+	  * Saves a screenshot of the whole screen in the .png format 
+	  */
+    public static void saveScreenshot(){
+   	 
+   	 /* Creates a new file with the next index number in function of the existing files of the same name('screenshot[number]') */
+        try{
+            FileHandle fh;
+            do{
+                fh = new FileHandle("Screenshots/screenshot" + screenshotNumber++ + ".png");
+            }while (fh.exists());
+            Pixmap pixmap = getScreenshot(0, 0, Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), true);
+            PixmapIO.writePNG(fh, pixmap);
+            pixmap.dispose();
+        }catch (Exception e){
+      	  System.err.println(e.getMessage());
+        }
+    }
+    
+    /**
+     * Gets a pixmap from the frame buffer.
+     * @param x XCoord
+     * @param y YCoord
+     * @param w Width
+     * @param h Heigth 
+     * @param yDown True if we want the screenshot flipped vertically, false if we don't
+     * @return The resulting pixmap 
+     */
+    private static Pixmap getScreenshot(int x, int y, int w, int h, boolean yDown){
+        final Pixmap pixmap = ScreenUtils.getFrameBufferPixmap(x, y, w, h);
+        
+        if(effects){
+      	  manipulate(pixmap);
+        }
+        
+        if (yDown) {
+            // Flip the pixmap upside down
+      	   ByteBuffer pixels = pixmap.getPixels();
+            int numBytes = w * h * 4;
+            byte[] lines = new byte[numBytes];
+            int numBytesPerLine = w * 4;
+            for (int i = 0; i < h; i++) {
+                pixels.position((h - i - 1) * numBytesPerLine);
+                pixels.get(lines, i * numBytesPerLine, numBytesPerLine);
+            }
+            pixels.clear();
+            pixels.put(lines);
+            pixels.clear();
+        }
+
+        return pixmap;
+    }
+    
+    /**
+     * Manipulates a pixmap with the conversions specified in this class
+     * @param pm The pixmap to be manipulated
+     */
+    private static void manipulate(Pixmap pm){
+   		 Color temp = new Color();
+          float avg;
+          for (int i = 0; i < pm.getWidth(); i++) {
+   			for (int j = 0; j < pm.getHeight(); j++) {
+   		      Color.rgba8888ToColor(temp, pm.getPixel(i, j));
+   		      if(grayScale){
+   		      	avg = (temp.r + temp.g + temp.b)/3;
+	   		      temp.r = avg;
+	   		      temp.g = avg;
+	   		      temp.b = avg;
+   		      }
+   		      pm.setColor(temp);
+   		      pm.drawPixel(i, j);
+   			}
+          }
+    }
+    
+    
+    /**
+     * Sets the gray scale flag to true or false 
+     */
+    public static void setGrayScale(){
+   	 effects = true;
+   	 grayScale = true;
+    }
+    
+    /**
+     * Checks if the capturer is set to take grayscaled screenshots
+     * @return the value of the grayScale member
+     */
+    public static boolean checkGrayScale(){
+   	 return grayScale;
+    }
+    
+    /**
+     * Disables grayscale mode
+     */
+    public static void disableGrayScale(){
```

Comment:
I'd expect `setGrayscale` to take a boolean to enable or disable, and there to be no `disableGrayscale` method. The `checkGrayscale` method should probably be called `isGrayscale`.